unix/w0vncserver/wayland/Desktop.cxx

```diff
@@ -0,0 +1,164 @@
+/* Copyright 2025 Adam Halim for Cendio AB
+ *
+ * This is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307,
+ * USA.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <assert.h>
+#include <unistd.h>
+#include <sys/mman.h>
+
+#include <glib.h>
+#include <wayland-client.h>
+#include <wayland-client-core.h>
+#include <wayland-client-protocol.h>
+
+#include <core/LogWriter.h>
+#include <rfb/VNCServerST.h>
+
+#include "../w0vncserver.h"
+#include "objects/Display.h"
+#include "objects/Output.h"
+#include "objects/Seat.h"
+#include "objects/VirtualPointer.h"
+#include "objects/VirtualKeyboard.h"
+#include "GWaylandSource.h"
+#include "PixelBuffer.h"
+#include "Desktop.h"
+
+using namespace wayland;
+
+static core::LogWriter vlog("WaylandDesktop");
+
+#define BUTTONS 9
+
+Desktop::Desktop(GMainLoop* loop_)
+  : server(nullptr), oldButtonMask(0), pb(nullptr), loop(loop_),
+    display(nullptr), seat(nullptr), waylandSource(nullptr),
+    virtualPointer(nullptr), virtualKeyboard(nullptr)
+{
+  assert(available());
+
+  display = new Display();
+  output = new Output(display);
+  seat = new Seat(display, std::bind(&Desktop::setLEDState,
+                                     this, std::placeholders::_1));
+}
+
+Desktop::~Desktop()
+{
+  delete pb;
+  delete waylandSource;
+  delete virtualPointer;
+  delete virtualKeyboard;
+  delete seat;
+  delete output;
+  delete display;
+}
+
+void Desktop::init(rfb::VNCServer* vs)
+{
+  server = vs;
+}
+
+void Desktop::start()
+{
+  std::function<void()> desktopReadyCb = [this]() {
+    virtualPointer = new VirtualPointer(display, seat);
+    virtualKeyboard = new VirtualKeyboard(display, seat);
+
+    server->setPixelBuffer(pb);
+    server->setLEDState(virtualKeyboard->getLEDState());
+  };
+
+  pb = new PixelBuffer(display, output, server, desktopReadyCb);
+
+  waylandSource = new GWaylandSource(display);
+  waylandSource->attach(g_main_loop_get_context(loop));
+}
+
+void Desktop::stop()
+{
+  server->setPixelBuffer(nullptr);
+
+  delete virtualKeyboard;
+  virtualKeyboard = nullptr;
+
+  delete waylandSource;
+  waylandSource = nullptr;;
+
+  delete virtualPointer;
+  virtualPointer = nullptr;
+
+  delete pb;
+  pb = nullptr;
+}
+
+void Desktop::pointerEvent(const core::Point& pos,
+                                  uint16_t buttonMask)
```

Comment:
Something wonky with the indentation here. Haven't checked if other places also got damaged in the rename.